web_poet/overrides.py

```diff
@@ -0,0 +1,305 @@
+            :func:`~.web_poet.overrides.consume_modules`.
+        """
+        if consume:
+            consume_modules(*_as_list(consume))
+
+        return list(self.values())
+
+    def search_overrides(self, **kwargs) -> List[OverrideRule]:
+        """Returns any :class:`OverrideRule` that has any of its attributes
+        match the rules inside the registry.
+
+        Sample usage:
+
+        .. code-block:: python
+
+            rules = registry.search_overrides(use=ProductPO, instead_of=GenericPO)
+            print(len(rules))  # 1
+
+        """
+
+        # Short-circuit operation if "use" is the only search param used, since
+        # we know that it's being used as the dict key.
+        if set(["use"]) == kwargs.keys():
+            rule = self.get(kwargs["use"])
+            if rule:
+                return [rule]
+            return []
+
+        getter = attrgetter(*kwargs.keys())
+
+        def matcher(rule: OverrideRule):
+            attribs = getter(rule)
+            if not isinstance(attribs, tuple):
+                attribs = tuple([attribs])
+            if list(attribs) == list(kwargs.values()):
+                return True
+            return False
+
+        results = []
+        for rule in self.get_overrides():
+            if matcher(rule):
+                results.append(rule)
+        return results
+
+
+def _walk_module(module: str) -> Iterable:
+    """Return all modules from a module recursively.
+
+    Note that this will import all the modules and submodules. It returns the
+    provided module as well.
+    """
+
+    def onerror(err):
+        raise err  # pragma: no cover
+
+    spec = importlib.util.find_spec(module)
+    if not spec:
+        raise ImportError(f"Module {module} not found")
+    mod = importlib.import_module(spec.name)
+    yield mod
+    if spec.submodule_search_locations:
+        for info in pkgutil.walk_packages(
+            spec.submodule_search_locations, f"{spec.name}.", onerror
+        ):
+            mod = importlib.import_module(info.name)
+            yield mod
+
+
+def consume_modules(*modules: str) -> None:
+    """This recursively imports all packages/modules so that the ``@handle_urls``
+    annotation are properly discovered and loaded.
+
+    Let's take a look at an example:
+
+    .. code-block:: python
+
+        # FILE: my_page_obj_project/load_rules.py
+
+        from web_poet import default_registry, consume_modules
+
+        consume_modules("other_external_pkg.po", "another_pkg.lib")
+        rules = default_registry.get_overrides()
+
+    For this case, the :class:`~.OverrideRule` are coming from:
+
+        - ``my_page_obj_project`` `(since it's the same module as the file above)`
+        - ``other_external_pkg.po``
+        - ``another_pkg.lib``
+
+    If the ``default_registry`` had other ``@handle_urls`` annotations outside
+    of the packages/modules listed above, then the corresponding
+    :class:`~.OverrideRule` won't be returned.
```

Comment:
Again, this is only true if they're not imported in some other way. 

I think it may be fine to just explain that consume_module imports modules recursively.

Reply:
Updated this in https://github.com/scrapinghub/web-poet/pull/27/commits/37e5ba1a3696f629801478fe78ea1729164f39df